Source/DafnyCore/CoverageReporter/CoverageReport.cs

```diff
@@ -0,0 +1,119 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using System.IO;
+using System.Linq;
+using Microsoft.Dafny;
+
+namespace DafnyCore.CoverageReporter; 
+
+public class CoverageReport {
+
+  private static int nextUniqueId = 0;
+
+  // INVARIANT: CoverageSpans are sorted within each list by the position of the StartToken
+  private readonly Dictionary<string, List<CoverageSpan>> labelsByFile;
+  public readonly string Name; // the name to assign to this coverage report
+  public readonly string Units; // the units of coverage (plural). This will be written in the coverage report table.
+  private readonly int uniqueId = nextUniqueId++;
+  public string UniqueId => "." + (uniqueId == 0 ? "" : (uniqueId + ".")); // to add as postfix to files
```

Comment:
When writing to disk, I always create a new directory, so overwriting shouldn't be a problem. I added the option to use a user-supplied suffix in my latest commit, so the set up should be better now. Even so, the numerical suffix is still necessary because when you merge arbitrary coverage reports from disk, there must be some way to differentiate between them, and they might have been created with the same user-provided suffix.

Reply:
> Even so, the numerical suffix is still necessary because when you merge arbitrary coverage reports from disk, there must be some way to differentiate between them, and they might have been created with the same user-provided suffix.

Sorry I don't understand this. Why can't you merge files with the same filename that are in different directories?

Reply:
When serializing a report, I create an `.html` file for each `.dfy` file in the sources. When serializing several reports, I therefore have several `.html` files for a single original `.dfy` file and so I have to differentiate between them in some way. By default, I will use the user-provided suffix indicating what kind of coverage is being reported by each `.html` file. But if these suffixes are equal as well, I propose to use a numerical index.